rsconnect/http_support.py

```diff
@@ -75,7 +75,12 @@ def _create_ssl_connection(host_name, port, disable_tls_check, ca_data):
     """
     if ca_data is not None and disable_tls_check:
         raise ValueError("Cannot both disable TLS checking and provide a custom certificate")
-    _, _, proxyHost, proxyPort = _get_proxy()
+
+    no_proxy = os.environ.get("no_proxy", os.environ.get("NO_PROXY", "#"))
```

Comment:
Why is the default value `'#'`?

Reply:
It won't ever match a hostname. Since the check is `endswith`, an empty string would be interpreted as matching all hosts.